src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -85,6 +87,19 @@ public static class SqlServerBaseRegistrationExtensions
                 .AsSelf()
                 .AsImplementedInterfaces();
 
+            switch (config.ConnectionType)
+            {
+                case SqlServerConnectionType.ManagedIdentity:
+                    services.AddSingleton<ISqlConnection, ManagedIdentitySqlConnection>();
+                    services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
+                    break;
+                case SqlServerConnectionType.WindowsIntegratedAuth:
```

Comment:
Is there any benefit to having this type since it's identical to `SqlUserNameAndPassword`? Is there a better name we could use to where there's only one of them?

Reply:
I agree. Both of them completely rely on connection string. I am lost for names though, completely open for suggestions.